lib/sensu/client/process.rb

```diff
@@ -455,27 +455,33 @@ def setup_standalone
       # Setup the Sensu client JSON socket, for external check result
       # input. By default, the client socket is bound to localhost on
       # TCP & UDP port 3030. The socket can be configured via the
-      # client definition, `:socket` with `:bind` and `:port`. The
-      # current instance of the Sensu logger, settings, and transport
+      # client definition, `:socket` with `:bind` and `:port`. Users can opt-out
+      # of using the TCP and UDP socket by setting `:enabled` to `false`.
+      # The current instance of the Sensu logger, settings, and transport
       # are passed to the socket handler, `Sensu::Client::Socket`. The
       # TCP socket server signature (Fixnum) and UDP connection object
       # are stored in `@sockets`, so that they can be managed
       # elsewhere, eg. `close_sockets()`.
       def setup_json_socket
         options = @settings[:client][:socket] || Hash.new
+        options[:enabled] ||= true
```

Comment:
Went with being explicit with the default but, if we'd rather I don't define this and check for the `nil?` instead, happy to switch.

Reply:
well following the existing pattern they do not check for `nil` and relies on defaults explicitly defined. I am :+1: on following the existing so I'd keep this and probably remove the conditional for `nil?`.